### PR TITLE
use template generate verifier.sol

### PR DESCRIPTION
**Pull request by xander42280**

### CODE REVIEW

File: testdata/mips/verifier.sol [hunk 1]
```diff
@@ -165,80 +165,80 @@ contract Verifier {
...
+        vk.gamma = Pairing.G2Point([uint256(4034795186869585120726589505662029195557861221653564273495998423601908830447), uint256(20042412918216602477550252825282063673019129406406673950976927898189556222221)], [uint256(17753313997110339421545663744727316684743608585913435600878477397674386347569), uint256(13215682004533589101102087044659372738424392632340918945254050743007224239453)]);
+        vk.delta = Pairing.G2Point([uint256(4312536892412607205246661494021023434710070905109206464045317216602273929990), uint256(20906595892652445275497483540707104193122424431346069768057806877559188349807)], [uint256(14522251126508817766161768944618892831358540985823309140492358589982581899258), uint256(14423812391745536634957010004318866620857657478006162190558100968639311942020)]);
+        vk.gamma_abc = new Pairing.G1Point[](66);
+        vk.gamma_abc[0] = Pairing.G1Point(uint256(4075695733072685169937636834237077667137159710258881045644857905464408005177), uint256(9972586854211772148424630562602477296659468076599166767757485372968755281541));
+        vk.gamma_abc[1] = Pairing.G1Point(uint256(0), uint256(0));
```
Comment (eigmax) [inline testdata/mips/verifier.sol:174]: Can we use a loop to assign the abc?

File: verifier/verifier.sol.tmpl [hunk 2]
```diff
@@ -0,0 +1,210 @@
+// This file is MIT Licensed.
```
Comment (eigmax) [inline verifier/verifier.sol.tmpl:1]: we can use tpl or template as the file suffix.

File: verifier/verifier.sol.tmpl [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        vk_x = Pairing.addition(vk_x, vk.gamma_abc[0]);
+        vk_x = Pairing.addition(vk_x, p_c);
+
+    if(!Pairing.pairingProd4(
+             proof.a, proof.b,
+             Pairing.negate(vk_x), vk.gamma,
+             Pairing.negate(proof.c), vk.delta,
+             Pairing.negate(vk.alpha), vk.beta)) return 1;
+
+    return 0;
```
Comment (eigmax) [inline verifier/verifier.sol.tmpl:196]: plz format this file.

File: verifier/sol_verifier.go [hunk 4]
```diff
@@ -51,7 +51,7 @@ type VerifierProof struct {
 // ContractMetaData contains all meta data concerning the Contract contract.
```
Comment (eigmax) [inline verifier/sol_verifier.go:51]: what's the purpose of ContractMetaData?

File: testdata/mips/verifier.sol (hunk 1 above)
Comment (zxz2801) [reply testdata/mips/verifier.sol:174]: Here is an example, in reality, dynamic data is filled in through templates

File: verifier/sol_verifier.go (hunk 4 above)
Comment (zxz2801) [reply verifier/sol_verifier.go:51]: This is abigen generated, If using hardhat deployment, there is no need to rely on this file anymore

File: verifier/verifier.sol.tmpl (hunk 2 above)
Comment (zxz2801) [reply verifier/verifier.sol.tmpl:1]: This is a template file

File: verifier/sol_verifier.go (hunk 4 above)
Comment (xander42280) [reply verifier/sol_verifier.go:51]: This is abigen generated, If using hardhat deployment, there is no need to rely on this file anymore

File: testdata/mips/verifier.sol (hunk 1 above)
Comment (xander42280) [reply testdata/mips/verifier.sol:174]: This is a template filled with dynamic data

**Review by eigmax — APPROVED** [review]

**Pull request merged**